### PR TITLE
Create Account View layout 

### PR DESCRIPTION
**Pull request by bluwave**
refs: MBL-14197
affects: Parent
release note: none

test
1. enable parentQRCodePairing on parent app
2. enable studentQRCodePairing on student app
3. open student app > settings > pair with observer > should see qr code
4. scan qr code in student app w/ device that has parent app
5. parent app should open to sign up page
6. sign up
7. try to login with new user.

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/6rul4w6) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F6rul4w6)
teacher: [Link](https://rebrand.ly/8vns29p) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F8vns29p)
parent: [Link](https://rebrand.ly/b1qxm47) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fb1qxm47)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Parent

[MBL-14197](https://instructure.atlassian.net/browse/MBL-14197)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Core/Core/Views/KeyboardTransitioning.swift:61:38: Unused Closure Parameter Violation: Unused parameter "completed" in a closure should be replaced with _. (unused_closure_parameter)
    Linting 'LoadingViewController.swift' (696/756)
    Linting 'CoreWebViewController.swift' (697/756)
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Profile/Settings/APIPairingCode.swift [hunk 1]
```diff
@@ -61,22 +63,52 @@ public struct PostAccountUserRequest: APIRequestable {
...
             let name: String
             let initial_enrollment_type: String
+            let terms_of_use: Bool = true
         }
         let pseudonym: Pseudonym
         let pairing_code: PairingCode
         let user: User
     }
+
+    public func urlRequest(relativeTo baseURL: URL, accessToken: String?, actAsUserID: String?) throws -> URLRequest {
```
Comment (bluwave) [inline Core/Core/Profile/Settings/APIPairingCode.swift:84]: Implementing this method so we can use a custom `baseURL` for creating acccount that may different than what env.api is set to.

Comment (vanwagonet) [reply Core/Core/Profile/Settings/APIPairingCode.swift:84]: I don't think this is necessary to get the desired effect. You can have an absolute url in the `path`, which allows you to point to a different domain, and have an explicit null authorization in `headers` to exclude sending the token. `GetMobileVerifyRequest` does this.

Comment (cobbal) [reply Core/Core/Profile/Settings/APIPairingCode.swift:84]: What are the advantages of this over using an absolute url with the usual setup? (it looks like this does a lot, so there may well be something different)

Comment (bluwave) [reply Core/Core/Profile/Settings/APIPairingCode.swift:84]: This worked well, thanks for the suggestion.

File: Parent/Parent/Create Account/CreateAccountViewController.swift [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+class CreateAccountViewController: UIViewController, ErrorViewController {
+    @IBOutlet weak var scrollView: UIScrollView!
+    @IBOutlet weak var stackView: UIStackView!
+    @IBOutlet weak var name: CreateAccountRow!
+    @IBOutlet weak var email: CreateAccountRow!
+    @IBOutlet weak var password: CreateAccountRow!
+    @IBOutlet weak var createAccountButton: DynamicButton!
+    @IBOutlet weak var termsAndConditionsLabel: DynamicLabel!
+    @IBOutlet weak var alreadyHaveAccountLabel: DynamicLabel!
+    var keyboardSpace: NSLayoutConstraint = NSLayoutConstraint()
```
Comment (vanwagonet) [inline Parent/Parent/Create Account/CreateAccountViewController.swift:31]: this is usually the scrollView.bottomAnchor to safeArea.bottomAnchor. I think this is making the scrollView not shrink when the keyboard appears, so you can't easily scroll to the next form field.

**Pull request merged**